Lib/test/libregrtest/main.py

```diff
@@ -318,9 +318,9 @@ def run_tests_sequentially(self, runtests):
             result = self.run_test(test_name, runtests, tracer)
 
             # Unload the newly imported modules (best effort finalization)
-            for module in sys.modules.keys():
+            for module in list(sys.modules):
```

Comment:
I don't understand this code. Why do we have to unload test modules?

If it's important to unload test modules, I would suggest to move this code inside single.py, to make a more consistent behavior for the different ways to run tests.

Reply:
Do you ask why the code of libregrtest is so complicated? :man_shrugging:

I think that it only makes sense in the single-process run. As well as removing the testing module before reloading in `_load_run_test()`. In the multi-process run it is only a waste of time. As well as significant part of `save_env` (except cleaning up external resources like filesystem).